devlib/module/cgroups.py

```diff
@@ -180,13 +205,19 @@ def tasks(self, cgroup):
         tasks = {}
         for task in entries:
             tid = task.split(',')[0]
+            if filter_tid and not re.search(filter_tid, tid):
```

Comment:
Just to be a bit more sure it's a "sort of regexp, should we use instead:
```python
if type(filter_tid) != str and re.search(filter_tid, tid):
```

Reply:
You can skip the check on type if you do the sanity check at the beginning...

Reply:
Done: sanity check at the start of the function